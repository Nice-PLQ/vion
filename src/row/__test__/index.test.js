import { mount } from '@vue/test-utils';
import VRow from '..';
import VCol from '../../col';

describe('Row', () => {
  test('渲染', () => {
    const wrapper = mount({
      template: `
        <v-row justify="center" align="center" gutter="8">
          <v-col span="6">span: 6</v-col>
          <v-col span="6" offset="16">span: 6</v-col>
          <v-col span="6">span: 6</v-col>
        </v-row>
      `,
    }, {
      global: {
        components: { VRow, VCol }
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    const { classList } = wrapper.find('.v-row').element;
    expect(classList).toContain('v-row_justify-center');
    expect(classList).toContain('v-row_align-center');
    expect(classList).toContain('v-row_justify-center');

    const cols = wrapper.findAll('.v-col');
    for (let i = 0; i < cols.length; i++) {
      expect(cols[i].element.classList).toContain('v-col_span-6');
      if (i === 0) {
        expect(cols[i].element.style.paddingLeft).not.toBe('8px');
      } else {
        expect(cols[i].element.style.paddingLeft).toBe('8px');
      }
    }
  });
});
