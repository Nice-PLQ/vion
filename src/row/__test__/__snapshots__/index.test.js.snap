// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row 渲染 1`] = `
<div
  class="v-row v-row_justify-center v-row_align-center v-row_no-wrap"
>
  
  <div
    class="v-col v-col_span-6"
  >
    
    span: 6
    
  </div>
  <div
    class="v-col v-col_span-6 v-col_offset-16"
    style="padding-left: 8px;"
  >
    
    span: 6
    
  </div>
  <div
    class="v-col v-col_span-6"
    style="padding-left: 8px;"
  >
    
    span: 6
    
  </div>
  
</div>
`;
